src/Loaders/LoadingSpinner.tsx

```diff
@@ -0,0 +1,76 @@
+/** @jsx jsx */
+import React from "react";
+import { jsx, keyframes } from "@emotion/core";
+import { colors } from "../colors";
+
+export type Size = "large" | "medium" | "small" | "tiny" | "icon";
+interface Props {
+  darkBackground?: boolean;
+  size?: Size;
+}
+
+// The whole animation is exactly 5 seconds long.
+// Each rotation is 450 degrees, with a deceleration at
+// 420 degrees, and a reduced speed between 420-450.
+// 1st rotation: 60 deg and 90 deg
+// 2nd rotation: 150 deg and 180 deg
+// 3rd rotation: 240 deg and 270 deg
+// 4th rotation: 330 deg and 360 deg (restart loop)
+const SPIN = keyframes`
+  25% { transform: rotate(450deg) }
+  50% { transform: rotate(900deg) }
+  75% { transform: rotate(1350deg) }
+  100% { transform: rotate(1800deg) }
+`;
+
+const SIZE_MAP = {
```

Comment:
Typescript already catches the error you mentioned above since its type stays as the specific map and doesnt get reduced to `{ [key: string]: number }`

Reply:
Well that's not exactly true. TypeScript does not catch the above error when we add a new `Size` and forget to add it to `SIZE_MAP`. Somewhere else in the code will have an error, but that's only because you chose to not enforce your typing where it matters. In other words, there is a TypeScript error, but that's by a happy coincidence. I'm asking you to make `SIZE_MAP` explicitly defined as "this is a number for every `Size`", which is _not_ what you're doing now. This will show exactly what our intent was.

## Before

```diff
diff --git a/src/Loaders/LoadingSpinner.tsx b/src/Loaders/LoadingSpinner.tsx
index a9632aa..6279c5b 100644
--- a/src/Loaders/LoadingSpinner.tsx
+++ b/src/Loaders/LoadingSpinner.tsx
@@ -3,7 +3,7 @@ import React from "react";
 import { jsx, keyframes } from "@emotion/core";
 import { colors } from "../colors";
 
-export type Size = "large" | "medium" | "small" | "tiny" | "icon";
+export type Size = "huge" | "large" | "medium" | "small" | "tiny" | "icon";
 export type Theme = "light" | "dark";
 interface Props {
   /**
```

```shell
➜  space-kit git:(jgzuke/add-loading-spinner) ✗ npx tsc --noEmit
src/Loaders/LoadingSpinner.tsx:68:21 - error TS7053: Element implicitly has an 'any' type because expression of type 'Size' can't be used to index type '{ large: number; medium: number; small: number; tiny: number; icon: number; }'.
  Property 'huge' does not exist on type '{ large: number; medium: number; small: number; tiny: number; icon: number; }'.

68   const pixelSize = SIZE_MAP[size];
                       ~~~~~~~~~~~~~~


Found 1 error.
```

## After

This is in contrast to making the requested change, which will show exactly what we're expecting instead of relying on an implicit error elsewhere:

```diff
diff --git a/src/Loaders/LoadingSpinner.tsx b/src/Loaders/LoadingSpinner.tsx
index a9632aa..e306927 100644
--- a/src/Loaders/LoadingSpinner.tsx
+++ b/src/Loaders/LoadingSpinner.tsx
@@ -3,7 +3,7 @@ import React from "react";
 import { jsx, keyframes } from "@emotion/core";
 import { colors } from "../colors";
 
-export type Size = "large" | "medium" | "small" | "tiny" | "icon";
+export type Size = "huge" | "large" | "medium" | "small" | "tiny" | "icon";
 export type Theme = "light" | "dark";
 interface Props {
   /**
@@ -36,7 +36,7 @@ const SPIN = keyframes`
   100% { transform: rotate(1800deg) }
 `;
 
-const SIZE_MAP = {
+const SIZE_MAP: Record<Size, number> = {
   large: 90,
   medium: 64,
   small: 48,

```

```zsh
➜  space-kit git:(jgzuke/add-loading-spinner) ✗ npx tsc --noEmit
src/Loaders/LoadingSpinner.tsx:39:7 - error TS2741: Property 'huge' is missing in type '{ large: number; medium: number; small: number; tiny: number; icon: number; }' but required in type 'Record<Size, number>'.

39 const SIZE_MAP: Record<Size, number> = {
         ~~~~~~~~


Found 1 error.
```

Reply:
I didn't say this in the comment so there's no fault on your part; I was hoping you'd respond to this message instead of just making the change and requesting a new review. 

It's important to me that you understand why I'm requesting this change and why I think this makes future code changes safer. Why? If you are looking for the same things, then you'll catch places where I might have missed strong type safety in my PRs. A rising tide lifts all ships!

Reply:
I didn't say this in the comment so there's no fault on your part; I was hoping you'd respond to this message instead of just making the change and requesting a new review. 

It's important to me that you understand why I'm requesting this change and why I think this makes future code changes safer. Why? If you are looking for the same things, then you'll catch places where I might have missed strong type safety in my PRs. A rising tide lifts all ships!